packages/docs/src/content/docs/reference/faq.md

```diff
@@ -0,0 +1,326 @@
+---
+title: FAQ
+date: 2024-08-21
+sidebar:
+  hidden: true
+---
+
+## Introduction
+
+Knip finds and removes unused files, dependencies and exports. As a "kitchen
+sink" in the npm ecosystem, it creates a comprehensive module and dependency
+graph of your project.
+
+The JavaScript ecosystem has a vast amount of frameworks and tools, and even
+more ways to configure those. Files and dependencies can be referenced in many
+ways, not just through import statements. This FAQ is an attempt to provide some
+perspective on why certain things work the way they do.
+
+## Why isn't Knip a ESLint plugin?
+
+Linters like ESLint analyze files separately, while Knip lints projects as a
+whole.
+
+Knip requires a full dependency graph to find things that are not referenced
+across the project. Creating this graph is not a trivial task and it seems no
+such tool exists today, even more so when it comes to monorepos.
+
+## Why doesn't Knip use an existing module resolver?
+
+Runtimes like Node.js provide `require.resolve` and `import.meta.resolve`.
+TypeScript comes with module resolution built-in. More module resolvers are out
+there and bundlers are known to use or come with module resolvers. None of them
```

Comment:
Also, some people customize module resolutions through tools like Webpack or Babel. (Not sure it's worth mentioning, but that's the first thing I thought of)